### DISCUSSION TITLE
Dynamically injecting beans into Quarkus

### DISCUSSION THREAD

**Discussion by masecla22** [quarkusio/quarkus#39895] [Q&A]
Hello!

I'm working on a SignalR implementation for Quarkus, and I'm looking to implement this as a library (rather than an extension)

That being said, the pattern I'm aiming for is to register endpoints dynamically both for HTTP and Websockets. 

For now I'm exclusively dealing with websockets:

```kotlin
abstract class SignalRHub(
    val transports: Array<TransportType> = arrayOf(
        TransportType.WEBSOCKET
        // TODO: Add the other transports
    )
) : Extension {
    @Path("/negotiate")
    @POST
    fun negotiate() : String {
        // Logic
    }

    @PostConstruct
    fun init() {
        // If transports contains websocket,
        // inject a handler for websockets at that the given path
    }

}
```

As far as I can tell, undertow-sockets had exactly what I needed in the form of the ServerContainer which allowed for dropping in place of new endpoints. 

How would I go about achieving something like this for Quarkus Websockets / HTTP?

PS:

This is what an expected implementation would possibly look like

```kotlin
@Path("/chat")
@ApplicationScoped
class Dumper : SignalRHub() {
}
```

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
If you are using JAX-RS annotations, then depending on `quarkus-rest` should be enough
